code/controllers/master.dm

```diff
@@ -224,7 +235,8 @@ var/CURRENT_TICKLIMIT = TICK_LIMIT_RUNNING
 	var/sleep_delta = 0
 	var/list/subsystems_to_check
 	//the actual loop.
-	while (1)
+	looping = TRUE
+	while (running)
```

Comment:
To reduce overhead I intentionally reduced the number of non-proc level vars accessed in this loop.